src/Lib.hs

```diff
@@ -75,19 +67,14 @@ getTile :: (MonadIO m, MonadError ServantErr m, MonadReader ServerState m)
         => Text -> Integer -> Integer -> Integer -> m (Headers '[Header "Last-Modified" String] ByteString)
 getTile l z x y = do
   geoJson <- getJson' l z x y
-  s <- ask
-  let pp = _pluginDir s
-  let startTime = _startTime s
+  pp <- asks _pluginDir
+  lastModified <- getLastModified l
   eet <- liftIO $ tileReturn geoJson pp
   case eet of
-    Left e -> pure $ addHeader startTime $ encodeUtf8 e
-    Right tile -> pure $ addHeader startTime tile
-    where tileReturn geoJson' pp' = fromGeoJSON defaultTileSize
-                                  geoJson'
-                                  l
-                                  pp'
-                                  (ZoomLevel z)
-                                  (GoogleTileCoords x y)
+    Left e -> pure $ addHeader lastModified $ encodeUtf8 e
```

Comment:
Return type says we do.  It does beg the question if you get an intermittent error - like If you can't get the database - you should update the modified header so that everything gets invalidated.